packages/teams-js/eslint-rules/eslint-plugin-recommend-const-enums/recommendConstEnums.js

```diff
@@ -0,0 +1,29 @@
+module.exports = {
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description: 'Recommend using const enums if possible to minimize bundle size',
+      category: 'Best Practices',
+      recommended: true,
+    },
+    fixable: 'code',
+    schema: [],
+  },
+
+  create: function (context) {
+    return {
+      TSEnumDeclaration: function (node) {
+        const enumName = node.id.name;
+        const enumType = node.const ? 'const' : 'regular';
+
+        if (enumType === 'regular') {
+          context.report({
+            node: node,
+            message: `Please consider if you can use a const enum for ${enumName} to minimize bundle size. If not, add "/* eslint-disable-next-line recommend-const-enums/recommend-const-enums */" to the line above to disable this warning.`,
```

Comment:
>Please consider if you can use a const enum for ${enumName} to minimize bundle size. [](http://example.com/codeflow?start=22&length=84)

Is there a documentation page we can point people at to learn about the impact and ramifications about using `const` enum? E.g., I learned a bunch from everything you've written about it recently, but any random person who makes a change probably won't know anything when they see this warning and if they bother to do any research they likely won't find anything as comprehensive as what you've shared.

Even if you just document what you've learned on a new page on [the wiki](https://github.com/OfficeDev/microsoft-teams-library-js/wiki), and then link to that, that would be great. In addition to explaining the impact on package size and why, providing guidance like:

"You should default to using a const enum unless you know that you need one of these things:
- Reverse mapping blah blah"
- etc.
- etc.
"

Reply:
I added a wiki page [here](https://github.com/OfficeDev/microsoft-teams-library-js/wiki/Notes-on-Enum-Usage), and linked to it in the warning message. Give the wiki page a read and let me know if you think there's anything I should add/remove/change etc.

Reply:
Looks great; thanks for writing a good explanation!